server/core/runtime/pre_workflow_hook_runner.go

```diff
@@ -22,18 +22,19 @@ func (wh DefaultPreWorkflowHookRunner) Run(ctx models.WorkflowHookCommandContext
 
 	baseEnvVars := os.Environ()
 	customEnvVars := map[string]string{
-		"BASE_BRANCH_NAME": ctx.Pull.BaseBranch,
-		"BASE_REPO_NAME":   ctx.BaseRepo.Name,
-		"BASE_REPO_OWNER":  ctx.BaseRepo.Owner,
-		"COMMENT_ARGS":     strings.Join(ctx.EscapedCommentArgs, ","),
-		"DIR":              path,
-		"HEAD_BRANCH_NAME": ctx.Pull.HeadBranch,
-		"HEAD_COMMIT":      ctx.Pull.HeadCommit,
-		"HEAD_REPO_NAME":   ctx.HeadRepo.Name,
-		"HEAD_REPO_OWNER":  ctx.HeadRepo.Owner,
-		"PULL_AUTHOR":      ctx.Pull.Author,
-		"PULL_NUM":         fmt.Sprintf("%d", ctx.Pull.Num),
-		"USER_NAME":        ctx.User.Username,
+		"BASE_BRANCH_NAME":     ctx.Pull.BaseBranch,
+		"BASE_REPO_NAME":       ctx.BaseRepo.Name,
+		"BASE_REPO_OWNER":      ctx.BaseRepo.Owner,
+		"COMMENT_ARGS":         strings.Join(ctx.EscapedCommentArgs, ","),
+		"DIR":                  path,
+		"HEAD_BRANCH_NAME":     ctx.Pull.HeadBranch,
+		"HEAD_COMMIT":          ctx.Pull.HeadCommit,
+		"HEAD_REPO_NAME":       ctx.HeadRepo.Name,
+		"HEAD_REPO_OWNER":      ctx.HeadRepo.Owner,
+		"PULL_AUTHOR":          ctx.Pull.Author,
+		"PULL_NUM":             fmt.Sprintf("%d", ctx.Pull.Num),
+		"TF_APPEND_USER_AGENT": fmt.Sprintf("Atlantis (+%s; %s; %s)", ctx.Pull.URL, ctx.Pull.HeadCommit, ctx.User.Username),
```

Comment:
Why would we use the `ctx.User.Username` here instead of the `ctx.Pull.Author` ?

Is the `ctx.User.Username` the one that is running the terraform workflow?

Can we also include the command whether it was a `plan` or `apply`?

Reply:
I _think_ `ctx.User` is the user who actually triggered atlantis which may or may not be the same as the pull request author. I think it's more useful to have that actor because you can use the included PR link to find the author. I don't know if the `command` argument to this function is necessarily plan/apply because you can run custom commands via workflows so there may not be a clean way to accomplish that without refactoring

Reply:
Could you test this out?

Reply:
If you could add a test too that would be nice!

Reply:
@lyoung-confluent friendly ping to help get this in the next release

Reply:
@lyoung-confluent friendly ping